### ISSUE TITLE
Web server stack

### ISSUE THREAD

**Issue by jotapea** [CNTRPRTY/xcpdev-ssr#1]
This is the place to discuss the alternatives to implement the server (side) rendered version of xcpdev.

The main requirement is having FULL control of the server. This IS the software being build, a web server to be run by the fednode, that provides a Counterparty data explorer out of the box.

Personally, I would be fine with doing a simple minimalist vanilla js (on express) version. A continuation of https://github.com/CNTRPRTY/xcpdev-ssr/commit/6eb8b16bcf2a2b63c8d06f29d2c0c1c8e3e507be.

But, this might be too simplistic, as any kind of dynamism in the generated HTML+JavaScript will complicate things quickly. Also, building HTML from plain strings is not developer friendly. But, this option is more long term stable, and less likely to break from outdated dependencies.

Explaining this to give a sense of what we should be looking for. What is the next best minimalist alternative to vanilla js?

No alternative:
- Just do vanilla JavaScript (accept tradeoffs for full control)

Library only, without framework (help in dev experience):
- [Preact](https://preactjs.com/)
- [React](https://react.dev/)

Framework (are we in control of the server?):
- [Next.js](https://nextjs.org/)
- [Remix](https://remix.run/)
- [Deno Fresh](https://fresh.deno.dev/)
- [Vue](https://vuejs.org/)

What else is out there that I am missing? Frontend development is not my expertise so looking forward to this discussion...

**Comment by blocklack** [comment]
Vue 3  has a low learning curve. and it has server side rendering included in its core.

**Comment by JavierCervilla** [comment]
I recommend from my experience nextjs or deno fresh, loving deno fresh now I'm testing it, and we can reuse code from the stamps indexer explorer if you decide to go for this way

**Comment by jotapea** [comment]
@blocklack Vue seems like a full framework, thus added to that list.

@JavierCervilla Having both share code would be pretty cool actually.

**Comment by Chriton** [comment]
I recommend nextJs and I am willing to do this as I told you after I finish with the complete redesign of the spa one.
You also need to keep in mind the pool of people willing to work on this and finish it not just start it.
If it's angular or vue, I'm not into that.

@jotapea  - Regarding your question  - _Framework (are we in control of the server?)_ - not sure what you mean by that. Of course you own the code and so you can deploy it on whatever server you want

**Comment by jotapea** [comment]
@Chriton yes is just that, being super clear we need to be able to deploy it ourselves (without breaking any terms).

Angular is discarded. And good to know you are not into Vue.
